Liberty olo with rh sso

rumanaHaque opened the pull request:
Pull request to get the blog content into staging. Please let me know of any comments and or updates. Thanks.

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+// change the "IMAGE CAPTION" to a couple words of what the image is
+// // // // // // // //
+
+Do you want to secure your Open Liberty Application deployed using Open Liberty Operator using OIDC? 
```
This feels a little sudden and out of the blue. I don't think it's needed, as the paragraph underneath provides a better introduction. I would remove this sentence.

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+
+Do you want to secure your Open Liberty Application deployed using Open Liberty Operator using OIDC? 
+
+Security is paramount in today's world, and it is important to ensure your cloud applications are protected. A common way to do that is to configure Single Sign On for your application such as GitHub, Google, Facebook or OpenID Connect (OIDC). This blog will show you step by step how to configure your Liberty application deployed in OpenShift to use a specific type of OIDC - RedHat Single SignOn  ...
```
Would be nice in this section, after the second sentence, to touch upon why single sign on is useful or why it's a common way to sign on? Why do people choose it? I.e. why would the reader want to do it and therefore want to read this blog?

gljrobinson replied:
Before the 3rd sentence about what this blog will teach, it would also be nice to have an intro as to what RH-SSO is and why people may choose to use it or want to use it. This, combined with the suggested sentence about why SSO is good, would together provide a solid foundation/introduction as to why a reader would need or want this blog, or why they'd want to do what we're suggesting in this blog. This feels missing at the moment.

gljrobinson replied:
I would perhaps suggest splitting this paragraph in two, at the point where we start to describe what this blog will show. The first part of the paragraph needs more introduction to some of the technologies this blog covers, which will make it longer. Then the second part would be easier to read as it's own paragraph which states out exactly what we're teaching in this blog (almost like a prose version of a contents list).

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+---
+layout: post
+title: "TITLE"
```
Need to change file name, to affect date of publication.

rumanaHaque replied:
Will change to 2024-08-10

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+
+== Set up the example application project from Social Media Guide
+
+Before we get into the details of configuring Security, we will first set up an example app. For this example, we will have a look at the application used in the Open Liberty guide -  link:/guides/social-media-login.html[Authenticating users through social media providers]
```
Perhaps suggest changing this paragraph slightly to something like:
"Before we get into the details of configuring Security, we first need to set up an example application to apply this security to. For this example, we will use the application provided in the Open Liberty guide -..."

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+
+//https://openliberty.io/guides/social-media-login.html
+
+However, instead of using GitHub to authenticate as shown in the guide above, we will first deploy this application, and use OIDC using RH-SSO to authenticate.
```
I think this paragraph could perhaps be a little confusing for anyone who has not read the guide and knows that this other guide uses GitHub for SSO. I was exacting readers to simply go to the github project, clone the repo and use the app but not to read the guide's instructions, so I'm not sure if this is needed? If it is, perhaps it could be rephrased to something like:
"In the Authenticating users through social media providers guide, GitHub is used for application authentication through the Open Liberty Social Media Login feature. However, in this blog, instead of directly using social platforms to directly authenticate with our application, we will use OIDC through RH-SSO."

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+cd guide-social-media-login
+----
+
+We will work within the start/ directory for this demonstration. Before we run the application, we will need to make some changes to the server.xml given in the link:https://github.com/OpenLiberty/guide-social-media-login/start/src/main/liberty/config/server.xml[server.xml] location.
```
I think "start/" needs to be formatted so that we know it's referencing the directory structure.

gljrobinson replied:
May be nice, to add a "However," to the start of the second sentence. So, "However, before we run...".

gljrobinson replied:
Don't think we need "Given in the server.xml location". The link for server.xml should be moved to the reference prior to this and then this part can be deleted.

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+
+We will work within the start/ directory for this demonstration. Before we run the application, we will need to make some changes to the server.xml given in the link:https://github.com/OpenLiberty/guide-social-media-login/start/src/main/liberty/config/server.xml[server.xml] location.
+
+Here is the updated server.xml that you have to use:
```
Maybe change this to:
"Here is the updated server.xml you'll need to use"

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+
+----
+
+Basically, you will have to add the feature socialLogin-1.0 to the feature list, and add the ports, and comment out the config about adding the keystore and truststore.
```
I think this tone is too casual for written content like this blog.
Maybe change this instead to something like: "This edited server.xml provides the feature socialLogin-1.0 to your application's feature list, and add the required ports. The configuration previously provided for the keystore and truststore is now commented out as this is no longer needed when running the application with the Open Liberty Operator, as we will do in this example."

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+
+Basically, you will have to add the feature socialLogin-1.0 to the feature list, and add the ports, and comment out the config about adding the keystore and truststore.
+
+After updating the server.xml file, run the application using the command:
```
Could be nice to add a bit to this to make the reader feel more engaged. So perhaps something like: "After updating the server.xml file, you're now ready to run the application. Use the following commands:"

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+mvn liberty:run
+----
+
+You should see the line that says 
```
Add to this to explain why or when they should see this. Does this line show that the application is up or ready or built?

gljrobinson replied:
This comment still stands, please address this. E.g., Once the application has finished building, you should see....

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+mvn package
+----
+
+This command builds a target/guide-social-login.war archive. We can now include this WAR in a container image that will be used to deploy this application in OCP.
```
This order feels like it may be the wrong way round. We jumped into building a WAR file and giving them the command for this before explaining why this is necessary (which comes after). Would be good to re-order this, so that we explain why we need the WAR file first and then get them to use the command to build it.

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+
+For the application to be deployed on Open Shift using the Open Liberty Operator, it must first be containerized using the Open Liberty image.
+
+For this example, we will use an official image from the IBM Container Registry (ICR), `icr.io/appcafe/open-liberty:full-java17-openj9-ubi`, as the parent image. 
```
Combine this with the sentence above. It feels too much like a list this way, needs to feel more like prose as this is a blog and not a tutorial/guide. Needs to feel more like conversation and less like instructions.

gljrobinson replied:
This comment still stands.

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+
+For this example, we will use an official image from the IBM Container Registry (ICR), `icr.io/appcafe/open-liberty:full-java17-openj9-ubi`, as the parent image. 
+
+In the start/ directory create a Dockerfile with these contents
```
This didn't link with the sentence above. Same comment as before, it needs to feel more like prose I think. At the moment, it just feels like separated instructions without the info on why or how they link together.

gljrobinson replied:
Can we link this more to the sentences before it so it links more? Something like... "to containerize the image, in the start directory...."?

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+
+As a pre-requisite to configure this environment, you need to have access to an Open Shift cluster.
+
+Install the Open Liberty Operator (OLO) in your OCP cluster using this doc - 
```
Maybe change to: "If you don't already have the Open Liberty Operator (OLO) in your OCP cluster, visit this documentation to see how to do this."

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+
+== Installing Open Liberty Operator to the RedHat OpenShift cluster
+
+As a pre-requisite to configure this environment, you need to have access to an Open Shift cluster.
```
May be nice to link out to a piece of material that could help someone who doesn't already have an Open Shift cluster.

rumanaHaque replied:
I think that is out of scope for this blog.

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+
+
+
+== Installing and configuring RH-SSO (RedHat Single Sign-On) Operator in the OpenShift cluster
```
This section feels a little chaotic and hard to read. Is there a way we could make these instructions clearer through formatting or changing the structure? Not sure, but I find it hard to read/follow at the moment.

gljrobinson replied:
I think maybe some images or screenshots somewhere in the blog would help to break this up and add context and visual interest.

gljrobinson replied:
This section is hard to read at the moment as it's very instructional and not written as a blog. We need more context and chatty prose in this section. I'll try and give some examples at the start of this section. Please could you try and copy my format for the rest of this section?

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+
+[.img_border_light]
+image::img/blog/rh_social_media_logged_in.png[Social Media Logged in after Redirect,width=50%,align="center"]
+
```
Would maybe be nice to have a summary section here to summarise what the reader has achieve by following the instructions in this blog and why these technologies are useful when building cloud-native applications. One short paragraph would be enough here. The blog doesn't feel finished or rounded off at the moment.

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+
+== Create the olapp-sso secret 
+
+Create a new project - called `gsm-test`, and create a secret in that namespace - - (Workloads->Secrets->Create Secret), called `guide-social-media-login-olapp-sso`, using key `oidc-clientId` and value `gsmapp`.
```
could this have a screenshot to make it easier to visualise/understand?

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+
+The key name should be in this format <app-name>-olapp-sso. Use the same <app-name> as what you will use while deploying the applicaton using OLO.
+
+For e.g. in this setup, the <app-name> from the yaml file is `guide-social-media-login`, so the secret name is `guide-social-media-login-olapp-sso`
```
for e.g. doesn't make sense, needs to be changed to "For example, in the application used here in this blog, the <app-name>...."

gljrobinson replied:
If we're following the example app for this tutorial, can we just assume that it has to be this name and instruct them to use this instead of explaining that that's what it is for this blog? Make it more explicit?

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+
+Create a new project - called `gsm-test`, and create a secret in that namespace - - (Workloads->Secrets->Create Secret), called `guide-social-media-login-olapp-sso`, using key `oidc-clientId` and value `gsmapp`.
+
+The key name should be in this format <app-name>-olapp-sso. Use the same <app-name> as what you will use while deploying the applicaton using OLO.
```
Would be helpful in second sentence to make sure it's obvious that they must do this. So change to "You must use the same...."

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+
+Note the name of the application deployed is `guide-social-media-login`, the same name that was used when creating the secret above.
+
+Point to your image location in this section
```
Not sure where these commands are meant to be being used? In terminal? In Dockerfile? Is it clear to others?

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+
+point to the OpenID Endpoint Configuration that you configured while configuring the RH-SSO Operator.
+
+Here is the yaml file needed to deploy the application:
```
Ahh, just realized that these above are all elements that are in this yaml file below. This is confusing. Instead, provide yaml up front and then underneath explain aspects of it without duplication source code. This sticks with same format used above in this blog and also that used in guides too. At the moment it feels like the reader needs these pieces of source code twice.

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+----
+== Create the OIDC Client in RH-SSO
+
+Since we have already deployed the guide-social-media-login app using the Open Liberty operator, we can now complete the registration for the openid client.
```
Should openid be OpenID?

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+
+Since we have already deployed the guide-social-media-login app using the Open Liberty operator, we can now complete the registration for the openid client.
+
+Use this url - https://keycloak-rh-sso.apps.<cluster-name>/auth/admin/master/console/
```
Why? what does this URL allow us to access? Not clear, don't know why I'm doing this step.

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+Use this url - https://keycloak-rh-sso.apps.<cluster-name>/auth/admin/master/console/
+using credentials from the secret - credential-example-keycloak
+
+Click on Create to create a new client, with clientId as `gsmapp`. (The same value that you put in the secret created called `guide-social-media-login-olapp-sso`).
```
Feels again like abstract instructions. Can these different paragraphs be linked more together and have more prose to explain how I get from one to the other and why?
(e.g. No that we have used the credentials, we can now create a new client to enable this authentication" --> not sure if that's accurate, just wanted to give an example of the sort of thing I meant by this comment".

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+
+Enter the URL for Valid Redirect URIs. In the scenario with 'oidcLogin', the URL will be https://<app-name>-<namespace>.apps.<cluster-name>/ibm/api/social-login/redirect/oidc
+
+For my test, I put in this value
```
First time you've used "my" and it jars with the rest of the tenses used in this article. Also, what test? what are we testing? why? This seems disjointed from the rest of the instructions.

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+== Running the application, and logging in using OIDC
+
+
+Since all the configuration is complete, you are ready to run the application now.
```
Could make this more engaging to audience as it's a lot of instructions they've just followed and has probably taken quite a bit of time. Maybe something more like: "Congratulations! You've now completed all the required configuration. Now, you're ready to run the application. To do this...."

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,394 @@
+image::img/blog/rh_social_media_login.png[Social Media Login,width=50%,align="center"]
+
+
+Since I have already created the RH-SSO client for this application, when I click on the "Log In" button for this app, it will redirect me to the RH-SSO client, as shown below.
```
Use of I again, doesn't work with tenses used within rest of blog. Also isn't engaging for readers.

gljrobinson replied:
This seems to be the crux of the whole article (i.e. what we've been trying to achieve this whole time), but it feels like a throwaway side comment with the short length. I think this section should have more prose and more impact.

GraceJansen reviewed (commented):
Initial thoughts after first review

gljrobinson commented:
Target to complete this now end of Feb 2024

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,462 @@
+// // // // // // // //
+
+
+Security is paramount in today's world, and it is important to ensure your cloud applications are protected. A common way to do that is to configure Single Sign On for your application such as GitHub, Google, Facebook or OpenID Connect (OIDC). Single Sign-On (SSO) is a mechanism that allows users to authenticate themselves once and gain access to multiple applications or systems without the need  ...
```
```suggestion
Security is paramount in today's world, and it is important to ensure your cloud applications are protected. A common way to do that is to configure Single Sign On for your application such as GitHub, Google, Facebook or OpenID Connect (OIDC). Single Sign-On (SSO) is a mechanism that allows users to authenticate themselves once and gain access to multiple applications or systems without the need to re-enter their credentials every time. This improves productivity and reduces the number of attack surfaces by reducing the number of times users have to sign in and only requiring the use of one set of credentials. To pass authorization from a single sign-on (SSO) service to another cloud application, OAuth can be used. This can help to further improve the security of an application as user credentials can be better protected by storing a token, and not the credentials themselves in local storage, filesystem or cookies.
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,462 @@
+
+
+Security is paramount in today's world, and it is important to ensure your cloud applications are protected. A common way to do that is to configure Single Sign On for your application such as GitHub, Google, Facebook or OpenID Connect (OIDC). Single Sign-On (SSO) is a mechanism that allows users to authenticate themselves once and gain access to multiple applications or systems without the need  ...
+
```
Would be nice to have a sentence here to introduce OIDC and why someone may want to specifically use this type of single-sign on (i.e. why we felt the need to write this blog/why it's helpful). At the moment it seems to dive into assuming people want to use this specific type of single sign on, when previously in the paragraph above it was only listed as one of the many options.

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+
+Security is paramount in today's world, and it is important to ensure your cloud applications are protected. A common way to do that is to configure Single Sign On for your application such as GitHub, Google, Facebook or OpenID Connect (OIDC). Single Sign-On (SSO) is a mechanism that allows users to authenticate themselves once and gain access to multiple applications or systems without the need  ...
+
+This blog will show you step by step how to configure your Liberty application deployed in OpenShift to use a specific type of OIDC - RedHat Single SignOn (RH-SSO). Liberty application will be deployed using the Open Liberty Operator (OLO) in OpenShift, with special configuration to be able to connect to RH-SSO. Configure the RH-SSO to create a client specifically for your liberty application. On ...
```
May be worth commenting here why we've focused on RH-SSO - i.e.this is the default SSO for any applications running on openshift, or something like that so that people know why they may use this.

Second sentence also needs to start with The - so "The Liberty application will be...."

Might be worth a short sentence here to cover why we're using the Liberty operator - i.e. The Open Liberty Operator (OLO) is being used here to help simplify the deployment process. To find out more about OLO and the benefits it offers see our documentation...." or something like this.

This sentence doesn't make sense on it's own, don't understand how it links to previous sentences - "Configure the RH-SSO to create a client specifically for your liberty application".

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+
+== Overview of steps
+
+Here are the high level steps that you need to follow:
```
```suggestion
To show how to configure your Liberty application to use RH-SSO, we'll take a simple Java application running on Liberty, deploy it to OpenShift and configure it to use RH-SSO. Below is an overview of the steps we'll be taking to do this that you can follow:
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+
+Here are the high level steps that you need to follow:
+
+1) Get the application ready
```
```suggestion
1) <<prepareApp, Set up, build and containerize the sample application>>
```

gljrobinson replied:
I don't think we need the sub-bullets under the numbered steps - at the moment it makes this section quite hard to read and follow and takes up a lot of space in this blog. I think we can just link to each step directly, as most of them only have one sub-bullet anyway.

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+* <<buildApp, Build the updated application>>
+* <<containerize, Containerizing the application>>
+
+2) Configure the RedHat OpenShift cluster. As a pre-requisite to configure this environment, you need to have access to an Open Shift cluster.
```
```suggestion
2) <<installRHSSO, Configure the Red Hat OpenShift cluster to make use of RH-SSO>>
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+
+* <<installRHSSO, Installing and configuring RH-SSO (RedHat Single Sign-On) Operator>>
+
+3) Install the OpenLiberty Operator, and deploy the Application in the Open Shift cluster using Open Liberty Operator
```
```suggestion
3) <<deployApp, Deploy the application to the Open Shift cluster using the Open Liberty Operator>>
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+* <<installRHSSO, Installing and configuring RH-SSO (RedHat Single Sign-On) Operator>>
+
+3) Install the OpenLiberty Operator, and deploy the Application in the Open Shift cluster using Open Liberty Operator
+Before you can deploy the application, you first need to create an OCP secret for the app.
```
```suggestion
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+
+4) Create a new client in RH-SSO for this deployed application, so that you can use RH-SSO to login to the application.
+
+* <<createOIDCClient, Create the OIDC Client in RH-SSO>>
```
```suggestion
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+3) Install the OpenLiberty Operator, and deploy the Application in the Open Shift cluster using Open Liberty Operator
+Before you can deploy the application, you first need to create an OCP secret for the app.
+
+* <<createSecret, Create the olapp-sso secret>>
```
```suggestion
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+
+* <<createSecret, Create the olapp-sso secret>>
+
+* <<deployApp, Install Open Liberty Operator, and deploy the application to Open Shift using the Open Liberty Operator>>
```
```suggestion
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+
+* <<deployApp, Install Open Liberty Operator, and deploy the application to Open Shift using the Open Liberty Operator>>
+
+4) Create a new client in RH-SSO for this deployed application, so that you can use RH-SSO to login to the application.
```
```suggestion
4) <<createOIDCClient, Create a new OIDC client in RH-SSO.>>
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+
+1) Get the application ready
+
+* <<setUpApp, Set up the example application from the guide>>
```
```suggestion
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+1) Get the application ready
+
+* <<setUpApp, Set up the example application from the guide>>
+* <<buildApp, Build the updated application>>
```
```suggestion
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+
+* <<setUpApp, Set up the example application from the guide>>
+* <<buildApp, Build the updated application>>
+* <<containerize, Containerizing the application>>
```
```suggestion
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+
+2) Configure the RedHat OpenShift cluster. As a pre-requisite to configure this environment, you need to have access to an Open Shift cluster.
+
+* <<installRHSSO, Installing and configuring RH-SSO (RedHat Single Sign-On) Operator>>
```
```suggestion
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+
+* <<createOIDCClient, Create the OIDC Client in RH-SSO>>
+
+5) Run the application, and logging in will redirect you to use OIDC using RH-SSO
```
```suggestion
5) <<runApp, Run the application, and log in using OIDC and RH-SSO.>>
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+
+5) Run the application, and logging in will redirect you to use OIDC using RH-SSO
+
+* <<runApp, Running the application, and logging in using OIDC>>
```
```suggestion
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+* <<runApp, Running the application, and logging in using OIDC>>
+
+
+[#setUpApp]
```
```suggestion
[#prepareApp]
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+
+
+[#setUpApp]
+== Set up the example application project from Social Media Guide
```
```suggestion
== Set up the example application from our Social Media Guide
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+//https://openliberty.io/guides/social-media-login.html
+
+
+Start by cloning the link:https://github.com/OpenLiberty/guide-social-media-login.git[Git repository] for this guide and use the projects that are provided inside:
```
```suggestion
Start by cloning the link:https://github.com/OpenLiberty/guide-social-media-login.git[Git repository] for this guide:
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+cd guide-social-media-login
+----
+
+The guide "Authenticating users through social media providers" mentioned above uses GitHub for application authentication through the Open Liberty Social Media Login feature. However, in this blog, instead of directly using social platforms to  authenticate with our application, we will use OIDC through RH-SSO.
```
Is it just GitHub this guide uses? I thought it was multiple social platforms you could use to login?

rumanaHaque replied:
This guide was using only GitHub, but my blog will be using OIDC.

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+
+Our first task will be to run the application on our machine, before we attempt to put it in a container.
+
+We will work within the link:https://github.com/OpenLiberty/guide-social-media-login/start/[start/] directory for this demonstration. However, before we run the application, we will need to make some changes to the server.xml.
```
```suggestion
However, before we run the application, we will need to make some changes to the server.xml. First, navigate to the link:https://github.com/OpenLiberty/guide-social-media-login/start/[start/] directory. Update the server.xml to the following:
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+
+We will work within the link:https://github.com/OpenLiberty/guide-social-media-login/start/[start/] directory for this demonstration. However, before we run the application, we will need to make some changes to the server.xml.
+
+Here is the updated server.xml that you will need to use:
```
```suggestion
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+
+This edited server.xml provides the feature socialLogin-1.0 to your application's feature list, and adds the required ports. The configuration previously provided for the keystore and truststore is now commented out as this is no longer needed when running the application with the Open Liberty Operator, as we will do in this example.
+
+[#buildApp]
```
```suggestion
```

gljrobinson replied:
don't need this anymore as we've combined these into one to simplify the content section.

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+This command builds a target/guide-social-login.war archive.
+Later on, you can use this WAR file in a container image that will be used to deploy this application in OCP.
+
+For now, run the application in your machine.
```
Combine this with the sentence below to make one paragraph for these, will make it easier to read as at the moment there's a lot of white space and individual lines.

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+RUN configure.sh
+
+----
+Ensure that you have these ENV values set to true.
```
A why is needed here --> why do we need to have these ENV values set to true? What do they do? Why are they needed?

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+COPY --chown=1001:0  target/guide-social-login.war /config/apps
+
+
+# This script will add the requested XML snippets and gow image to be fit-for-purpose
```
gow image?

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+By specifying ARG SEC_SSO_PROVIDERS="oidc", you are telling the configuration that the SSO provider you will be using is OIDC.
+
+
+Build the application image using the Docker File shown above, and upload to a repository of your choice (for e.g. dockerhub or artifactory), and note the image location so that you can use it later on for deploying this application to OpenShift using the Open Liberty Operator (OLO)
```
```suggestion
Build the application image using the Docker File shown above and upload to a repository of your choice (for e.g. dockerhub or artifactory). You'll need to make a note of the image location so that you can use it later on for deploying this application to OpenShift using the Open Liberty Operator (OLO).
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+
+Build the application image using the Docker File shown above, and upload to a repository of your choice (for e.g. dockerhub or artifactory), and note the image location so that you can use it later on for deploying this application to OpenShift using the Open Liberty Operator (OLO)
+
+
```
Can we add a sentence here to summarise that they have now successfully set up the sample application and can now move onto the next setp of installing and configuring their openshift cluster for us to deploy this too? This will help to round off this section and make a clear distinction between the steps we laid out at the start of this blog.

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+[#installRHSSO]
+== Installing and configuring RH-SSO (RedHat Single Sign-On) Operator in the OpenShift cluster
+
+Install the Red Hat Single Sign-On Operator to the cluster using these instructions:
```
```suggestion
In this section we'll walk through the steps necessary to correctly set up our OpenShift cluster so that we can make use of RH-SSO. 

1. Step 1 - Install RH-SSO

The first step we'll need to take is to install the Red Hat Single Sign-On Operator to the cluster. This will allow us to easily install RH-SSO. When installing this operator, ensure that it is installed in the namespace - "rh-sso". To do this, follow the instructions provided in the Red Hat documentation:
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+
+https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.6/html/server_installation_and_configuration_guide/operator#doc-wrapper
+
+Install the Operator in the namespace - "rh-sso".
```
```suggestion
```

gljrobinson commented on posts/2023-08-23-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,458 @@
+After installing the RH-SSO Operator, create a KeyCloak instance using the defaults.
+After creating the KeyCloak instance, you should be able to access the keycloak by looking at the routes.
+
+The route should be in this format - https://keycloak-rh-sso.apps.<cluster_name>
```
```suggestion

2. Step 2 - Create a KeyCloak instance

After installing the RH-SSO Operator, create a KeyCloak instance using the default values provided. After creating the KeyCloak instance, you should be able to access the KeyCloak by looking at the routes. The route should be in this format - https://keycloak-rh-sso.apps.<cluster_name>
```

GraceJansen reviewed (changes requested):
Hi Rumana, tried to get through as much of this as I could today but have run out of time. Wanted to send what I've reviewed over to you so far so yu can take a look at it and then I can always review more after.

gljrobinson commented:
It's looking great @rumanaHaque, thank you for all your work and effort on this and for making the changes suggested. There is only one thing missing from this blog ow and that is a summary/conclusion. This can just be a paragraph long, it doesn't have to be too much. It would be great if in this you could summarize what the learner has achieved and how it could be helpful to their own applications. If there's any other links/resources/guides/material that you think would be good for them to look at as further reading or the follow on from this, we can also include these as bulleted links in this section too.

rumanaHaque commented:
Hi @GraceJansen . I made all the changes that you had requested above, added a summary, and updated the file name to be posts/2024-08-10-liberty-olo-rh-sso.adoc instead of posts/2023-08-23-liberty-olo-rh-sso.adoc

Please take a look, and let me know if I need to make any other changes.

dmuelle commented on posts/2024-08-20-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,338 @@
+
+. To install RH-SSO in the `rh-sso` namespace, follow the instructions that are provided in the link:https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.6/html/server_installation_and_configuration_guide/operator#doc-wrapper[Red Hat Single Sign-On Operator documentation].
+
+. Create and log in to a KeyCloak instance.
```
I think this step needs more detail

rumanaHaque replied:
I can add more details by adding screen-shots. Will that help?
These are for installing the operator
<img width="896" alt="Screenshot 2024-08-15 at 4 39 37 PM" src="https://github.com/user-attachments/assets/ac8b8fa8-b3f0-43a1-ad8c-63db68523e28">


<img width="1854" alt="Screenshot 2024-08-15 at 4 41 49 PM" src="https://github.com/user-attachments/assets/bfa8e08c-3c19-4b03-b85c-58c57b808651">

dmuelle replied:
yes- i think screen shots of the console and a description of the steps would be helpful. Alternatively, if this is already documented in detail by redhat, we can provide a link to that specific section of their doc.

dmuelle commented on posts/2024-08-20-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,338 @@
+
+. Create and log in to a KeyCloak instance.
++
+After you install the RH-SSO Operator, create a KeyCloak instance that uses the default values. You can then access KeyCloak by looking at the routes. The route is in the `\https://keycloak-rh-sso.apps.<cluster_name>` format. You use this URL to log in to the KeyCloak instance. The credentials for logging in are in the `credential-example-keycloak` secret in the `rh-sso` namespace. Get the secret ...
```
> After you install the RH-SSO Operator, create a KeyCloak instance that uses the default values.
How? Is it the same process described in RH docs [11.3.2. Creating a Keycloak custom resource on OpenShift](https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html/server_installation_and_configuration_guide/operator#creating_a_keycloak_custom_resource_on_openshift) ? If so, we should link to it. I realize this section relies heavily on RH docs but we should make it as easy as we can for user to find the right info.

> You can then access KeyCloak by looking at the routes. 
Here too- how do they find the routes? 

> The route is in the `\https://keycloak-rh-sso.apps.<cluster_name>` format.

When I look at the "Keycloak custom resource Status" example in the RH doc, the route is not in the format we suggest here.

rumanaHaque replied:
This screenshot shows how to create the KeyCloak instance.

<img width="1333" alt="Screenshot 2024-08-15 at 4 46 24 PM" src="https://github.com/user-attachments/assets/58809c10-d12d-411c-aba7-74bd3aabb974">

And this screenshot here shows the routes for the rh-sso namespace in my cluster.

<img width="1710" alt="Screenshot 2024-08-15 at 5 05 53 PM" src="https://github.com/user-attachments/assets/dc6adecc-28c2-441f-8915-c34310bb1f60">

As you can see - it is in the format mentioned above:

`https://keycloak-rh-sso.apps.rhaque-gsm.cp.fyre.ibm.com`

dmuelle replied:
ok- either a link to the red hat doc section or screen shots and description would be helpful, and save the user time if they are not already familiar with RH SSO

dmuelle commented on posts/2024-08-20-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,338 @@
++
+After you install the RH-SSO Operator, create a KeyCloak instance that uses the default values. You can then access KeyCloak by looking at the routes. The route is in the `\https://keycloak-rh-sso.apps.<cluster_name>` format. You use this URL to log in to the KeyCloak instance. The credentials for logging in are in the `credential-example-keycloak` secret in the `rh-sso` namespace. Get the secret ...
+
+. Create a realm named `sso-realm`. Use the following URL to access this realm:
```
> Create a realm named `sso-realm`. 

Does this map to [Creating a realm custom resource](https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html/server_installation_and_configuration_guide/operator#realm-cr) in the RH doc?

rumanaHaque replied:
Yes - this document shows how to create a Realm in RH-SSO.

dmuelle commented on posts/2024-08-20-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,338 @@
+
+Next, we create a secret for the Open Liberty application. The key name for this secret is in the `<app-name>-olapp-sso` format. You must use the same `<app-name>` as the one you use to deploy the application with OLO. For example, in the application that is used in this post, the `<app-name>` from the YAML file is `guide-social-media-login`, so the secret name is `guide-social-media-login-olapp- ...
+
+. Create a project in your cluster called `gsm-test`.
```
> Create a project in your cluster called `gsm-test`.

How? is this documented on the RH side somewhere that we can point to?

rumanaHaque replied:
https://docs.openshift.com/container-platform/4.16/applications/projects/working-with-projects.html documents how to create a project.

dmuelle commented on posts/2024-08-20-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,338 @@
+* The name of the application is `guide-social-media-login`, the same name that was used when you created the previous secret.
+* The `oidc: discoveryEndpoint` must point to the OpenID endpoint configuration that you set when you configured the RH-SSO Operator.
+* The `applicationImage` value must point to your image location where you placed your application image, in a container registry like DockerHub or Artifactory.
+* The `pullSecret` value must be set to be able to access the container registry.
```
> The `pullSecret` value must be set to be able to access the container registry.

what should it be set to- is it the `guide-social-media-login-olapp-sso` secret they created in the previous step?

rumanaHaque replied:
No, this secret is different.
After you upload your container image (containing your application) to a container registry, besides pointing to the url that has the image, you also need to provide it the credentials to be able to access that registry.

In my case, I was using Artifactory to push the image to, and I have my pull-secret pointing to the credentials that I use to log into Artifactory.

This part is more generic, because we can use any type of container registry (such as dockerhub etc). But basically after you point to the image, you also have to provide the credentials that OCP will need to pull down that image.

dmuelle reviewed (changes requested):
Hi @rumanaHaque - I made some structural edits to the post to make it easier to follow. You can review the updated post at:

https://blogs-draft-openlibertyio.mqj6zf7jocq.us-south.codeengine.appdomain.cloud/blog/2024/08/20/liberty-olo-rh-sso.html

If you need to make ay further changes, remember to pull first as I worked from your working branch.

Also, I still have a few questions about steps that might need more details or links to external doc, which Ive noted in my review. I've set the tentative publish date to 8/20, but we can change that if we need more time to get it sorted. Thanks!

FYI @GraceJansen

rumanaHaque commented:
@dmuelle - I did not make any changes to the doc, but answered your questions as comments for each.
Please let me know if they make sense, and if so how I can incoporate them in my blog. 
Thanks.

dmuelle commented on posts/2024-08-20-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,338 @@
++
+After you install the RH-SSO Operator, create a KeyCloak instance that uses the default values. You can then access KeyCloak by looking at the routes. The route is in the `\https://keycloak-rh-sso.apps.<cluster_name>` format. You use this URL to log in to the KeyCloak instance. The credentials for logging in are in the `credential-example-keycloak` secret in the `rh-sso` namespace. Get the secret ...
+
+. Create a realm named `sso-realm`. Use the following URL to access this realm:
```
```suggestion
. link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html/server_installation_and_configuration_guide/operator#realm-cr[Create a realm] named `sso-realm`. Use the following URL to access this realm:
```

dmuelle commented on posts/2024-08-20-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,338 @@
+
+Next, we create a secret for the Open Liberty application. The key name for this secret is in the `<app-name>-olapp-sso` format. You must use the same `<app-name>` as the one you use to deploy the application with OLO. For example, in the application that is used in this post, the `<app-name>` from the YAML file is `guide-social-media-login`, so the secret name is `guide-social-media-login-olapp- ...
+
+. Create a project in your cluster called `gsm-test`.
```
```suggestion
. link:https://docs.openshift.com/container-platform/4.16/applications/projects/working-with-projects.html[Create a project] in your cluster called `gsm-test`.
```

dmuelle commented on posts/2024-08-20-liberty-olo-rh-sso.adoc:
```diff
@@ -0,0 +1,338 @@
+* The name of the application is `guide-social-media-login`, the same name that was used when you created the previous secret.
+* The `oidc: discoveryEndpoint` must point to the OpenID endpoint configuration that you set when you configured the RH-SSO Operator.
+* The `applicationImage` value must point to your image location where you placed your application image, in a container registry like DockerHub or Artifactory.
+* The `pullSecret` value must be set to be able to access the container registry.
```
```suggestion
* The `pullSecret` value must be set to the name of the secret that contains the credentials for your chosen container registry.
```

GraceJansen reviewed (approved):
LGTM
